diff --git a/include/wx/osx/iphone/chkconf.h b/include/wx/osx/iphone/chkconf.h
index aea925c..dc7bcf0 100644
--- a/include/wx/osx/iphone/chkconf.h
+++ b/include/wx/osx/iphone/chkconf.h
@@ -106,12 +106,12 @@
 #define wxUSE_STATUSBAR 0
 #define wxUSE_NATIVE_STATUSBAR 0
 #define wxUSE_ABOUTDLG 0
-#define wxUSE_STATLINE 0
+#define wxUSE_STATLINE 1
 #define wxUSE_COLLPANE 0
-#define wxUSE_STATBOX 0
-#define wxUSE_RADIOBTN 0
-#define wxUSE_RADIOBOX 0
-#define wxUSE_TOGGLEBTN 0
+#define wxUSE_STATBOX 1
+#define wxUSE_RADIOBTN 1
+#define wxUSE_RADIOBOX 1
+#define wxUSE_TOGGLEBTN 1
 
 #undef wxUSE_RICHTEXT
 #define wxUSE_RICHTEXT 0
@@ -147,15 +147,15 @@
 #define wxUSE_ADDREMOVECTRL 0
 #define wxUSE_ANIMATIONCTRL 0
 #define wxUSE_CALENDARCTRL 0
-#define wxUSE_COMBOCTRL 0
+#define wxUSE_COMBOCTRL 1
 #define wxUSE_ODCOMBOBOX 0
 #define wxUSE_BITMAPCOMBOBOX 0
-#define wxUSE_BMPBUTTON 0
-#define wxUSE_CHECKLISTBOX 0
+#define wxUSE_BMPBUTTON 1
+#define wxUSE_CHECKLISTBOX 1
 #define wxUSE_GRID 0
 #define wxUSE_LISTCTRL 0
-#define wxUSE_SPINBTN 0
-#define wxUSE_SPINCTRL 0
+#define wxUSE_SPINBTN 1
+#define wxUSE_SPINCTRL 1
 #define wxUSE_DATEPICKCTRL 0
 #define wxUSE_EDITABLELISTBOX 0
 #define wxUSE_FILEPICKERCTRL 0
@@ -201,9 +201,9 @@
 #define wxUSE_TOOLBOOK 0
 #define wxUSE_CHOICEDLG 0
 #define wxUSE_HELP 0
-#define wxUSE_PROGRESSDLG 0
+#define wxUSE_PROGRESSDLG 1
 #define  wxUSE_FONTDLG 0
-#define wxUSE_FILEDLG 0
+#define wxUSE_FILEDLG 1
 #define wxUSE_CHOICEDLG 0
 #define wxUSE_NUMBERDLG 0
 #define wxUSE_TEXTDLG 0
@@ -278,10 +278,7 @@
 #endif // wxUSE_GLCANVAS
 */
 
-#if wxUSE_COLOURDLG
-#undef wxUSE_COLOURDLG
-#define wxUSE_COLOURDLG 0
-#endif // wxUSE_COLOURDLG
+/* Orca-iOS-ipa: wxUSE_COLOURDLG kept enabled */
 
 // iphone has a toolbar that is a regular UIView
 
@@ -301,7 +298,7 @@
 
 #if wxUSE_INFOBAR
 #undef wxUSE_INFOBAR
-#define wxUSE_INFOBAR 0
+#define wxUSE_INFOBAR 1
 #endif
 
 #if wxUSE_FILE_HISTORY
@@ -369,6 +366,13 @@
 
 #endif // wxUSE_GUI
 
+
+#ifndef wxDEFAULT_BUTTON_MARGIN
+#define wxDEFAULT_BUTTON_MARGIN 4
+#endif
+#undef wxUSE_WEBVIEW
+#define wxUSE_WEBVIEW 1
+
 #endif
     /* _WX_OSX_IPHONE_CHKCONF_H_ */
 
