diff --git a/build/cmake/files.cmake b/build/cmake/files.cmake
index c163c87..25a9fb7 100644
--- a/build/cmake/files.cmake
+++ b/build/cmake/files.cmake
@@ -853,7 +853,7 @@ set(GUI_CMN_SRC
     src/generic/headerctrlg.cpp
     src/generic/infobar.cpp
     src/generic/listbkg.cpp
-    src/generic/logg.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/generic/logg.cpp>
     src/generic/markuptext.cpp
     src/generic/msgdlgg.cpp
     src/generic/numdlgg.cpp
@@ -1970,7 +1970,7 @@ set(OSX_COMMON_SRC
     src/osx/checkbox_osx.cpp
     src/osx/checklst_osx.cpp
     src/osx/choice_osx.cpp
-    src/osx/combobox_osx.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/osx/combobox_osx.cpp>
     src/osx/dnd_osx.cpp
     src/osx/gauge_osx.cpp
     src/osx/listbox_osx.cpp
@@ -1997,13 +1997,13 @@ set(OSX_COMMON_SRC
     src/osx/carbon/fontdlgosx.mm
     # other shared files
     src/osx/accel.cpp
-    src/osx/carbon/clipbrd.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/osx/carbon/clipbrd.cpp>
     src/osx/carbon/cursor.cpp
     src/osx/carbon/fontdlg.cpp
     src/osx/carbon/gdiobj.cpp
     src/osx/carbon/app.cpp
     src/osx/carbon/control.cpp
-    src/osx/carbon/dataobj.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/osx/carbon/dataobj.cpp>
     src/osx/carbon/dcclient.cpp
     src/osx/carbon/dcprint.cpp
     src/osx/carbon/dcscreen.cpp
@@ -2024,8 +2024,8 @@ set(OSX_COMMON_SRC
     src/generic/colrdlgg.cpp
     src/generic/dirdlgg.cpp
     src/generic/fdrepdlg.cpp
-    src/generic/filedlgg.cpp
-    src/generic/filepickerg.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/generic/filedlgg.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/generic/filepickerg.cpp>
     src/generic/fontdlgg.cpp
     src/generic/fontpickerg.cpp
     src/generic/listctrl.cpp
@@ -2235,6 +2235,16 @@ set(OSX_IPHONE_SRC
     src/osx/iphone/statbmp.mm
     src/osx/iphone/menuitem.mm
     src/osx/iphone/menu.mm
+    src/osx/statbox_osx.cpp
+    src/osx/statline_osx.cpp
+    src/osx/radiobut_osx.cpp
+    src/osx/radiobox_osx.cpp
+    src/osx/spinbutt_osx.cpp
+    src/osx/bmpbuttn_osx.cpp
+    src/generic/spinctlg.cpp
+    src/osx/iphone/filedlg.mm
+    src/osx/iphone/webview_stub.cpp
+    src/osx/iphone/mediactrl_stub.cpp
 )
 
 set(OSX_IPHONE_HDR
@@ -2257,8 +2267,8 @@ set(UNIV_SRC
     src/generic/colrdlgg.cpp
     src/generic/dirdlgg.cpp
     src/generic/fdrepdlg.cpp
-    src/generic/filedlgg.cpp
-    src/generic/filepickerg.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/generic/filedlgg.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/generic/filepickerg.cpp>
     src/generic/fontdlgg.cpp
     src/generic/fontpickerg.cpp
     src/generic/listctrl.cpp
@@ -2751,13 +2761,13 @@ set(UNIX_SOUND_SDL_SRC
 )
 
 set(AUI_CMN_SRC
-    src/aui/framemanager.cpp
-    src/aui/dockart.cpp
-    src/aui/floatpane.cpp
-    src/aui/auibook.cpp
-    src/aui/auibar.cpp
-    src/aui/tabmdi.cpp
-    src/aui/tabart.cpp
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/framemanager.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/dockart.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/floatpane.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/auibook.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/auibar.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/tabmdi.cpp>
+    $<$<NOT:$<BOOL:${IPHONE}>>:src/aui/tabart.cpp>
     src/xrc/xh_aui.cpp
     src/xrc/xh_auitoolb.cpp
 )
diff --git a/include/wx/colordlg.h b/include/wx/colordlg.h
index 264f4f7..db9b1fd 100644
--- a/include/wx/colordlg.h
+++ b/include/wx/colordlg.h
@@ -18,6 +18,9 @@
 
 #if defined(__WXMSW__) && !defined(__WXUNIVERSAL__)
     #include "wx/msw/colordlg.h"
+#elif defined(__WXMAC__) && defined(__WXOSX_IPHONE__)
+    #include "wx/generic/colrdlgg.h"
+    #define wxColourDialog wxGenericColourDialog
 #elif defined(__WXMAC__) && !defined(__WXUNIVERSAL__)
     #include "wx/osx/colordlg.h"
 #elif defined(__WXGTK__) && !defined(__WXUNIVERSAL__)
diff --git a/include/wx/filedlg.h b/include/wx/filedlg.h
index d3f0681..32bfdc8 100644
--- a/include/wx/filedlg.h
+++ b/include/wx/filedlg.h
@@ -281,6 +281,8 @@ wxSaveFileSelector(const wxString& what,
     #include "wx/msw/filedlg.h"
 #elif defined(__WXGTK__)
     #include "wx/gtk/filedlg.h"     // GTK+ > 2.4 has native version
+#elif defined(__WXMAC__) && defined(__WXOSX_IPHONE__)
+    #include "wx/osx/iphone/filedlg.h"
 #elif defined(__WXMAC__)
     #include "wx/osx/filedlg.h"
 #elif defined(__WXQT__)
