diff --git a/src/slic3r/Utils/Serial.cpp b/src/slic3r/Utils/Serial.cpp
index 4db1acc..5a2c1b4 100644
--- a/src/slic3r/Utils/Serial.cpp
+++ b/src/slic3r/Utils/Serial.cpp
@@ -34,11 +34,15 @@
 	#include "boost/nowide/convert.hpp"
 	#pragma comment(lib, "user32.lib")
 #elif __APPLE__
+	#include <TargetConditionals.h>
 	#include <CoreFoundation/CoreFoundation.h>
 	#include <CoreFoundation/CFString.h>
+	#if TARGET_OS_OSX
+	// IOKit serial services do not exist on iOS.
 	#include <IOKit/IOKitLib.h>
 	#include <IOKit/serial/IOSerialKeys.h>
 	#include <IOKit/serial/ioss.h>
+	#endif
 	#include <sys/syslimits.h>
 #endif
 
@@ -156,7 +160,7 @@ std::vector<SerialPortInfo> scan_serial_ports_extended()
 			output.emplace_back(std::move(port_info));
 		}
 	}
-#elif __APPLE__
+#elif __APPLE__ && TARGET_OS_OSX
 	// inspired by https://sigrok.org/wiki/Libserialport
 	CFMutableDictionaryRef classes = IOServiceMatching(kIOSerialBSDServiceValue);
 	if (classes != 0) {
@@ -306,7 +310,7 @@ void Serial::set_baud_rate(unsigned baud_rate)
 			}
 		};
 
-#if __APPLE__
+#if __APPLE__ && TARGET_OS_OSX
 		termios ios;
 		handle_errno(::tcgetattr(handle, &ios));
 		handle_errno(::cfsetspeed(&ios, baud_rate));
@@ -342,7 +346,7 @@ void Serial::set_baud_rate(unsigned baud_rate)
 		ios.c_cc[VTIME] = 1;
 		handle_errno(::ioctl(handle, TCSETS2, &ios));
 
-#elif __OpenBSD__
+#elif __OpenBSD__ || (defined(__APPLE__) && !TARGET_OS_OSX)
 		struct termios ios;
 		handle_errno(::tcgetattr(handle, &ios));
 		handle_errno(::cfsetspeed(&ios, baud_rate));
